vars/kola.groovy

```diff
@@ -116,15 +116,24 @@ def call(params = [:]) {
             ids += id
             runKola(id, 'run', "--parallel=${parallel} ${args} ${extraArgs}")
         } else {
-            // basic run
-            if (!params['skipBasicScenarios']) {
-                id = marker == "" ? "kola-basic" : "kola-basic-${marker}"
-                ids += id
-                def skipSecureBootArg = ""
+            // Check if the basic tests are registered kola tests or not.
+            def availableTests = shwrapCapture("kola list --json | jq -r '.[].Name'").split()
+            // This essentually checks if the running has https://github.com/coreos/coreos-assembler/pull/3652
+            def basicScenariosBuiltIn = "basic.nvme" in availableTests  
+            id = marker == "" ? "kola-basic" : "kola-basic-${marker}"
+            ids += id
+
+            if (basicScenariosBuiltIn) {
+                if (params['skipSecureBoot']) {
+                    // skip secureboot tests using kola's denylist
+                    runKola(id, 'run', "--denylist-test *.uefi-secure")
+                }
+            } else {
```

Comment:
This should be conditional on `!skipBasicScenarios`, right?

Reply:
It is, so if we basicScenariosBuiltIn or else we must not have basicScenariosBuiltIn ?

Reply:
Ugh sorry I read skipBasicScenarios and basicScenariosBuiltIn as the same var.

Reply:
I have updated the logic, sigh Idk how I dropped that conditional, sorry that this has been such a dumpster fire. Thank you for helping me through this.